cli/test/lib/exec/spawn_spec.js

```diff
@@ -171,6 +173,20 @@ describe('lib/exec/spawn', function () {
       })
     })
 
+    context('detects kill signal', function () {
+      it('exits with error on SIGKILL', function () {
+        this.spawnedProcess.on.withArgs('exit').yieldsAsync(null, 'SIGKILL')
+
+        return spawn.start('--foo')
+        .then(() => {
+          throw new Error('should have hit error handler but did not')
+        }, (e) => {
```

Comment:
the `.catch()` pattern is highly preferred over passing two arguments to `.then`

Reply:
well, in this case, `catch` would let false positive slide (we really want NOT to resolve). Ideally we would plug in https://www.chaijs.com/plugins/chai-as-promised/
```
return spawn.start('--foo').should.eventually.rejectWith(...)
```